build_distribution.sh

```diff
@@ -97,7 +97,7 @@ cd $RSS_HOME
 
 if [ -z "$JAVA_HOME" ]; then
   echo "Error: JAVA_HOME is not set, cannot proceed."
-  exit -1
```

Comment:
Why do we need to change this line?

Reply:
Hi @xianjingfeng 

there is a warning  reported by my IDE,  and I also see the return value is `1` in other file(e.g., `bin/start-shuffle-server.sh `)

`build_distribution.sh`
<img width="779" alt="image" src="https://github.com/apache/incubator-uniffle/assets/6762509/3c81f4b7-d5ae-4916-a9f5-d8f7ecc9bd1e">


`bin/start-shuffle-server.sh`
<img width="565" alt="image" src="https://github.com/apache/incubator-uniffle/assets/6762509/c2acec49-9dd8-4988-a642-ba70649e205e">

Reply:
> there is a warning reported by my IDE, and I also see the return value is `1` in other file(e.g., `bin/start-shuffle-server.sh `)

We can fix it in next pr.

Reply:
Sure, thanks for friendly reminder and I updated the part you mentioned.